Remove old agent plugin system

BrianJKoopman opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
This PR removes the old plugin system from its remaining usage in `ocsbow` and the `HostManagerAgent`. I also removed the docs for the plugin scan section of the [site config page](https://ocs.readthedocs.io/en/main/developer/site_config.html#agent-script-discovery), and any references to 'agent-paths' in the SCF, which aren't used by the new plugin system.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
The new plugin system has been around for just about two years now, introduced in [v0.10.0](https://github.com/simonsobs/ocs/releases/tag/v0.10.0). In `socs` we've continued to maintain both [ocs_plugin_so.py](https://github.com/simonsobs/socs/blob/main/socs/agents/`ocs_plugin_so.py`) (the old plugin file) alongside the new `plugin.py` and I'd like to stop doing that.

Resolves #322.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
I've run a small ocs network with two Host Managers, one for docker and one for agents running from the new plugin system. I've also run `ocsbow config plugins` and confirmed output. The output changed a little bit, just based on how the two plugin systems worked. The old system would show full paths to the modules like this:
```
$ ocsbow config plugins
ocs plugin detection
--------------------
Scanning.
  ... adding to path: /home/ocs/git/ocs/ocs/agents/
  ... adding to path: /home/ocs/git/socs/socs/agents/
Found:
  RegistryAgent        : /home/ocs/git/ocs/ocs/agents/registry/agent.py
  AggregatorAgent      : /home/ocs/git/ocs/ocs/agents/aggregator/agent.py
  HostManager          : /home/ocs/git/ocs/ocs/agents/host_manager/agent.py
  FakeDataAgent        : /home/ocs/git/ocs/ocs/agents/fake_data/agent.py
  InfluxDBAgent        : /home/ocs/git/ocs/ocs/agents/influxdb_publisher/agent.py
  BarebonesAgent       : /home/ocs/git/ocs/ocs/agents/barebones/agent.py
  ACUAgent             : /home/ocs/git/socs/socs/agents/acu/agent.py
  BlueforsAgent        : /home/ocs/git/socs/socs/agents/bluefors/agent.py
  CrateAgent           : /home/ocs/git/socs/socs/agents/smurf_crate_monitor/agent.py
  CryomechCPAAgent     : /home/ocs/git/socs/socs/agents/cryomech_cpa/agent.py
  FPGAAgent            : /home/ocs/git/socs/socs/agents/holo_fpga/agent.py
  FTSAerotechAgent     : /home/ocs/git/socs/socs/agents/fts_aerotech/agent.py
  HWPBBBAgent          : /home/ocs/git/socs/socs/agents/hwp_encoder/agent.py
  HWPPicoscopeAgent    : /home/ocs/git/socs/socs/agents/hwp_picoscope/agent.py
  ibootbarAgent        : /home/ocs/git/socs/socs/agents/ibootbar/agent.py
  LabJackAgent         : /home/ocs/git/socs/socs/agents/labjack/agent.py
  Lakeshore240Agent    : /home/ocs/git/socs/socs/agents/lakeshore240/agent.py
  Lakeshore336Agent    : /home/ocs/git/socs/socs/agents/lakeshore336/agent.py
  Lakeshore370Agent    : /home/ocs/git/socs/socs/agents/lakeshore370/agent.py
  Lakeshore372Agent    : /home/ocs/git/socs/socs/agents/lakeshore372/agent.py
  Lakeshore425Agent    : /home/ocs/git/socs/socs/agents/lakeshore425/agent.py
  LATRtXYStageAgent    : /home/ocs/git/socs/socs/agents/xy_stage/agent.py
  MagpieAgent          : /home/ocs/git/socs/socs/agents/magpie/agent.py
  MeinbergM1000Agent   : /home/ocs/git/socs/socs/agents/meinberg_m1000/agent.py
  PfeifferAgent        : /home/ocs/git/socs/socs/agents/pfeiffer_tpg366/agent.py
  PfeifferTC400Agent   : /home/ocs/git/socs/socs/agents/pfeiffer_tc400/agent.py
  PysmurfController    : /home/ocs/git/socs/socs/agents/pysmurf_controller/agent.py
  PysmurfMonitor       : /home/ocs/git/socs/socs/agents/pysmurf_monitor/agent.py
  RotationAgent        : /home/ocs/git/socs/socs/agents/hwp_rotation/agent.py
  ScpiPsuAgent         : /home/ocs/git/socs/socs/agents/scpi_psu/agent.py
  SmurfFileEmulator    : /home/ocs/git/socs/socs/agents/smurf_file_emulator/agent.py
  SmurfStreamSimulator : /home/ocs/git/socs/socs/agents/smurf_stream_simulator/agent.py
  SupRsync             : /home/ocs/git/socs/socs/agents/suprsync/agent.py
  SynaccessAgent       : /home/ocs/git/socs/socs/agents/synacc/agent.py
  SynthAgent           : /home/ocs/git/socs/socs/agents/holo_synth/agent.py
  TektronixAWGAgent    : /home/ocs/git/socs/socs/agents/tektronix3021c/agent.py
  ThorlabsMC2000BAgent : /home/ocs/git/socs/socs/agents/thorlabs_mc2000b/agent.py
  VantagePro2Agent     : /home/ocs/git/socs/socs/agents/vantagepro2/agent.py
  WiregridActuatorAgent : /home/ocs/git/socs/socs/agents/wiregrid_actuator/agent.py
  WiregridEncoderAgent : /home/ocs/git/socs/socs/agents/wiregrid_encoder/agent.py
  WiregridKikusuiAgent : /home/ocs/git/socs/socs/agents/wiregrid_kikusui/agent.py
```

The new one shows the modules like this:
```
$ OCS_CONFIG_DIR=$(pwd) ocsbow config plugins
ocs plugin detection
--------------------
Installed OCS Plugins: ['ocs', 'socs']
Found:
  RegistryAgent             : ocs.agents.registry.agent
  AggregatorAgent           : ocs.agents.aggregator.agent
  HostManager               : ocs.agents.host_manager.agent
  FakeDataAgent             : ocs.agents.fake_data.agent
  InfluxDBAgent             : ocs.agents.influxdb_publisher.agent
  BarebonesAgent            : ocs.agents.barebones.agent
  ACTiCameraAgent           : socs.agents.acti_camera.agent
  ACUAgent                  : socs.agents.acu.agent
  BlueforsAgent             : socs.agents.bluefors.agent
  CrateAgent                : socs.agents.smurf_crate_monitor.agent
  CryomechCPAAgent          : socs.agents.cryomech_cpa.agent
  FPGAAgent                 : socs.agents.holo_fpga.agent
  FlowmeterAgent            : socs.agents.ifm_sbn246_flowmeter.agent
  FTSAerotechAgent          : socs.agents.fts_aerotech.agent
  GeneratorAgent            : socs.agents.generator.agent
  Hi6200Agent               : socs.agents.hi6200.agent
  HWPBBBAgent               : socs.agents.hwp_encoder.agent
  HWPGripperAgent           : socs.agents.hwp_gripper.agent
  HWPPCUAgent               : socs.agents.hwp_pcu.agent
  HWPPicoscopeAgent         : socs.agents.hwp_picoscope.agent
  HWPPIDAgent               : socs.agents.hwp_pid.agent
  HWPPMXAgent               : socs.agents.hwp_pmx.agent
  HWPSupervisor             : socs.agents.hwp_supervisor.agent
  ibootbarAgent             : socs.agents.ibootbar.agent
  LabJackAgent              : socs.agents.labjack.agent
  Lakeshore240Agent         : socs.agents.lakeshore240.agent
  Lakeshore336Agent         : socs.agents.lakeshore336.agent
  Lakeshore370Agent         : socs.agents.lakeshore370.agent
  Lakeshore372Agent         : socs.agents.lakeshore372.agent
  Lakeshore425Agent         : socs.agents.lakeshore425.agent
  LATRtXYStageAgent         : socs.agents.xy_stage.agent
  MagpieAgent               : socs.agents.magpie.agent
  MeinbergM1000Agent        : socs.agents.meinberg_m1000.agent
  MeinbergSyncboxAgent      : socs.agents.meinberg_syncbox.agent
  PfeifferAgent             : socs.agents.pfeiffer_tpg366.agent
  PfeifferTC400Agent        : socs.agents.pfeiffer_tc400.agent
  PysmurfController         : socs.agents.pysmurf_controller.agent
  PysmurfMonitor            : socs.agents.pysmurf_monitor.agent
  ScpiPsuAgent              : socs.agents.scpi_psu.agent
  SmurfFileEmulator         : socs.agents.smurf_file_emulator.agent
  SmurfStreamSimulator      : socs.agents.smurf_stream_simulator.agent
  SmurfTimingCardAgent      : socs.agents.smurf_timing_card.agent
  SupRsync                  : socs.agents.suprsync.agent
  SynaccessAgent            : socs.agents.synacc.agent
  SynthAgent                : socs.agents.holo_synth.agent
  TektronixAWGAgent         : socs.agents.tektronix3021c.agent
  ThorlabsMC2000BAgent      : socs.agents.thorlabs_mc2000b.agent
  UCSCRadiometerAgent       : socs.agents.ucsc_radiometer.agent
  UPSAgent                  : socs.agents.ups.agent
  VantagePro2Agent          : socs.agents.vantagepro2.agent
  WiregridActuatorAgent     : socs.agents.wiregrid_actuator.agent
  WiregridEncoderAgent      : socs.agents.wiregrid_encoder.agent
  WiregridKikusuiAgent      : socs.agents.wiregrid_kikusui.agent
  WiregridTiltSensorAgent   : socs.agents.wiregrid_tiltsensor.agent
```

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

mhasself commented on ocs/agents/host_manager/agent.py:
```diff
@@ -317,16 +306,11 @@ def same_base_class(a, b):
-                    instance['agent_script'] = site_config.agent_script_reg[cls]
-                    instance['operable'] = True
                 else:
                     session.add_message(f'No plugin (nor launcher script) '
                                         f'found for agent_class "{cls}"!')
```
Stray reference to "launcher script".

mhasself reviewed (approved):
End of an era :)  Couple more hostmanager simplifications:
- in `_launch_instance`, if `management != docker` then you can assume `script == '__plugin__'` (and thus not check it, not if/else it, and build cmd in one statement).
- in `drivers.py`, change docs for "agent_script" to "The docker service_name, if docker-managed.  Otherwise, the string "__plugin__" to indicate it is host managed."

The name "agent_script" is now terrible ... probably better named "service_name".  But not crucial right now.  The docstring would be helpful though.

BrianJKoopman commented:
> Couple more hostmanager simplifications:
> 
> * in `_launch_instance`, if `management != docker` then you can assume `script == '__plugin__'` (and thus not check it, not if/else it, and build cmd in one statement).

Great, I simplified this in [ec5dde0](https://github.com/simonsobs/ocs/pull/407/commits/ec5dde0d15ff58970e4c2541dd6bb9c36727678a).

Makes me wonder if we still need to store `__plugin__` as the "agent_script". This was the only place that was used. But perhaps when that name gets changed it'll reflect its usage of storing the docker service name and will be removed anyway. I've left it in for now, seeing as it was in your docstring suggestion.

> * in `drivers.py`, change docs for "agent_script" to "The docker service_name, if docker-managed.  Otherwise, the string "**plugin**" to indicate it is host managed."

Done in [8a0a324](https://github.com/simonsobs/ocs/pull/407/commits/8a0a324d47bba0ab2a8f0c8280025e0fcc69d129). I also added an autodoc for the drivers to the HostManager agent docs page in [015506a](https://github.com/simonsobs/ocs/pull/407/commits/015506ac7643d77254d6098dca87d83fa544dd65).

> The name "agent_script" is now terrible ... probably better named "service_name". But not crucial right now. The docstring would be helpful though.

Yeah, agreed.

mhasself commented:
Thanks.

> Makes me wonder if we still need to store `__plugin__` as the "agent_script". This was the only place that was used. But perhaps when that name gets changed it'll reflect its usage of storing the docker service name and will be removed anyway. I've left it in for now, seeing as it was in your docstring suggestion.

I think it's still needed as a place-holder, but I might be wrong.  Was a little bit of work to figure out / replace it ... but the sort of thing I might deal with in next major HostManager update (if there is one).
